From 8cd915228bafb9792fd5d3e7e778db8ed51578d1 Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <fourdan@xfce.org>
Date: Sat, 1 Dec 2018 21:12:58 +0100
Subject: compositor: Shaded windows have no height

Bug: 14899

Trying to get the pixmap  with a height of zero will lead to an XError
and further trouble down the line.

Signed-off-by: Olivier Fourdan <fourdan@xfce.org>
---
 src/compositor.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/compositor.c b/src/compositor.c
index 500f68fd..81264550 100644
--- a/src/compositor.c
+++ b/src/compositor.c
@@ -3752,6 +3752,12 @@ compositorScaleWindowPixmap (CWindow *cw, guint *width, guint *height)
     }
     src_size = MAX (src_w, src_h);
 
+    /* Shaped windows have no height */
+    if (src_w == 0 || src_h == 0)
+    {
+        return None;
+    }
+
     /*/
      * Caller may pass either NULL or 0.
      * If 0, we return the actual unscalled size.
-- 
cgit v1.2.1

